src/Web/Sfa.Das.Sas.ApplicationServices/Handlers/SitemapHandler.cs

```diff
@@ -54,21 +53,56 @@ public SitemapResponse Handle(SitemapQuery message)
             };
         }
 
+        private IEnumerable<string> GetFrameworkDetailsInSeoFormat()
+        {
+            var frameworks = _getFrameworks.GetAllFrameworks();
+
+            return BuildFrameworkSitemap(frameworks);
+        }
+
+        private IEnumerable<string> GetStandardDetailsInSeoFormat()
+        {
+            var standards = _getStandards.GetAllStandards().Where(s => s.IsPublished);
+
+            return BuildStandardSitemap(standards);
+        }
+
         private IEnumerable<string> GetProviderDetailsInSeoFormat()
         {
             var providersExcludingEmployerProviders = _getProviders.GetAllProviders().Where(x => x.IsEmployerProvider == false);
-            var identifiers = BuildProviderSitemapFromProviders(providersExcludingEmployerProviders);
-            return identifiers;
+
+            return BuildProviderSitemapFromProviders(providersExcludingEmployerProviders);
+        }
+
+        private IEnumerable<string> BuildFrameworkSitemap(IEnumerable<Framework> frameworks)
+        {
+            return from framework in frameworks
+                   let title = EncodeTitle(framework)
+                   select GetSeoFormat(framework.FrameworkId, title);
+        }
+
+        private IEnumerable<string> BuildStandardSitemap(IEnumerable<Standard> standards)
+        {
+            return from standard in standards
+                   let title = EncodeTitle(standard)
+                   select GetSeoFormat(standard.StandardId, title);
         }
 
         private IEnumerable<string> BuildProviderSitemapFromProviders(IEnumerable<ProviderSummary> providers)
         {
-            foreach (var provider in providers)
-            {
-                var encodedProviderName = _urlEncoder.EncodeTextForUri(provider.ProviderName);
-                var urlLocElement = $@"{provider.Ukprn}/{encodedProviderName}";
-                yield return urlLocElement;
-            }
+            return from provider in providers
+                   let encodedProviderName = _urlEncoder.EncodeTextForUri(provider.ProviderName)
+                   select $@"{provider.Ukprn}/{encodedProviderName}";
+        }
+
+        private string GetSeoFormat(string id, string title)
+        {
+            return string.IsNullOrEmpty(title) ? $"{id}" : $"{id}/{title}";
+        }
+
+        private string EncodeTitle(IApprenticeshipProduct product)
+        {
+            return _urlEncoder.EncodeTextForUri(product.Title);
         }
 
         private XDocument CreateDocument(IEnumerable<string> items, string urlPlaceholder)
```

Comment:
I'm not sure this belongs here and is there a cleaner way to create the xml

Reply:
I have refactored this into a new class, but having looked at other examples online they are a similar quantity of code so not a massive benefit to change.